components/calendar/src/calendar/index.js

```diff
@@ -0,0 +1,105 @@
+import {
+    useDatePicker,
+    useResolvedDirection,
+} from '@dhis2/multi-calendar-dates'
+import { colors } from '@dhis2/ui-constants'
+import React, { useState } from 'react'
+import { CalendarProps } from './calendar-prop-types.js'
+import { CalendarTable } from './calendar-table.js'
+import { NavigationContainer } from './navigation-container.js'
+
+export const Calendar = ({
+    onDateSelect,
+    calendar,
+    date,
+    dir,
+    locale,
+    numberingSystem,
+    weekDayFormat,
+    timeZone,
+    width,
+    cellSize,
+}) => {
+    const [selectedDateString, setSelectedDateString] = useState(date)
+
+    const languageDirection = useResolvedDirection(dir, locale)
+
+    const styleOptions = {
+        backgroundColor: 'none',
+        chevronColor: colors.grey600,
+        dayHoverBackgroundColor: colors.grey200,
+        dayNamesColor: colors.grey700,
+        headerBackground: colors.grey050,
+        selectedDayBackgroundColor: colors.teal700,
+        wrapperBorderColor: colors.grey300,
+        width,
+        cellSize,
+    }
```

Comment:
This doesn't make sense to me. This object only has 2 dynamic properties. I see you've done this because the header and the table cells need similar styles. But these are different tags (`td` vs `th`). IMO it's fine to let both have their individual style rules which happen to be identical right now. If you don't like this, you could write a shared style sheet that targets both the `td` and the `th` tag.

In any case, the main point here is that we don't pass a style options object. Instead just pass the dynamic props directly to the components that need them, and use them in the styled-jsx there. This is a more transparent way of working imo.